lib/utils.js

```diff
@@ -23,20 +25,24 @@ module.exports = {
     };
   },
 
-  walkSync(dir, filelist) {
-    var files = fs.readdirSync(dir);
-    filelist = filelist || [];
-    files.forEach((file) => {
-      if (fs.statSync(path.join(dir, file)).isDirectory()) {
-        filelist = this.walkSync(path.join(dir, file), filelist);
-      }
-      else {
-        filelist.push(path.join(path.basename(dir), file));
-      }
-    });
-    return filelist;
+  walk(dir) {
+    return fs.readdir(dir)
+      .then((items) => {
+        return promiseMap(items, (item) => {
```

Comment:
Can you wrap the items in a promise function, push them in an array and use `Promise.all` ? I would prefer not to use another dependency just to map over an array.

Reply:
Sure, I'll map them to array of promises and then use `Promise.all`.